src/store/reducer.js

```diff
@@ -0,0 +1,52 @@
+
+const initialState = {
+    cart : [],
+    totalPrice: 0
+};
+
+const reducer = (state = initialState, action) => {
+
+    switch ((action.type)) {
+        case 'addProduct':
+            let newProduct = {id:action.id, name:action.name, price:action.price, description:action.description,
```

Comment:
There are a lot of variables here you can use const instead of let, each variables you dont change its value or initialize only once. check where.